### ISSUE TITLE
SchedXmlConfig:  Error 0x80070490: failed to find matching ElementId:

### ISSUE THREAD

**Issue by avivanoff** [wixtoolset/issues#7377]
WIX 4.0.0.0, Visual Studio 2022 17.5.3, HeatWave for VS2022 1.0.0.

A simple project that uses XmlConfig fails with `SchedXmlConfig:  Error 0x80070490: failed to find matching ElementId`. The same code works in WIX 3.

See attached [sample project](https://github.com/wixtoolset/issues/files/11179644/Package1.zip).

**Comment by barnson** [comment]
Please attach a verbose log.

**Comment by avivanoff** [comment]
@barnson, [attached](https://github.com/wixtoolset/issues/files/11180718/Package1.zip).

**Comment by robmen** [comment]
@avivanoff FYI, WiX v4 already shipped. It is very possible that if there is a bug here, the fix will go into WiX v5*. We'll triage this next week.

* I will be writing this down more formally.

**Comment by robmen** [comment]
Let's be real for a minute.

This is an Open Source project. We worked very hard over the last six months, to produce multiple release candidates for the community to test their key scenarios and flush out important issues. There was ample time to help find this bug had _you_ prioritized being part of the WiX community.

Similarly, I don't know you. You are not active in the community helping us make WiX better, and AFAICT, you are not a customer (if I'm wrong, please ignore all of this and [open a support ticket](https://support.firegiant.com/requests/new)). Asking strangers who put in a significant volunteer effort to immediately do more work for you is not an effective tactic for making friends and influencing people.

I know this is probably not a great time to learn this lesson how to interact with Open Source projects as a consumer. Hopefully, it encourages you to be more proactive in all of the projects you depend upon for free going forward. Alternatively, choose to become a customer and get your needs met as per a paid contract.

The next WiX Online Meeting is Tuesday, Apr 18, 2023, at 9:30 AM PDT.

**Comment by nirbar** [comment]
@avivanoff As a workaround you can use my open source and free WiX extension
nuget https://www.nuget.org/packages/PanelSwWixExtension4

Apply a XSLT to manipulate the xml file:
~~~
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:panelsw="http://schemas.panel-sw.co.il/wix/WixExtension">
  <Fragment>
    <Binary Id="xslt" SourceFile="$(sys.SOURCEFILEDIR)Test.xslt"/>
    <ComponentGroup Id="ExampleComponents" Directory="INSTALLFOLDER">
      <Component>
        <File Source="$(sys.SOURCEFILEDIR)test.xml">
          <panelsw:XslTransform BinaryKey="xslt"/>
        </File>
      </Component>
    </ComponentGroup>
  </Fragment>
</Wix>
~~~
Contact me if you need any help

**Comment by avivanoff** [comment]
@nirbar, looks like the fix is in and is supposed to be shipped as part of [4.0.1](https://github.com/wixtoolset/issues/milestone/20).
